driver-core/src/main/com/mongodb/connection/netty/NettyStreamFactoryFactory.java

```diff
@@ -99,6 +110,42 @@ public Builder eventLoopGroup(final EventLoopGroup eventLoopGroup) {
             return this;
         }
 
+        /**
+         * Sets the tuner for a {@linkplain SslContextBuilder#forClient() client-side} {@link SslContext io.netty.handler.ssl.SslContext},
+         * which overrides the standard {@link SslSettings#getContext()}.
+         * By default the tuner is {@code null} and {@link SslSettings#getContext()} is at play.
+         * <p>
+         * This option may be used as a convenient way to utilize
+         * <a href="https://www.openssl.org/">OpenSSL</a> as an alternative to the TLS/SSL protocol implementation in a JDK.
+         * To achieve this, specify {@link SslProvider#OPENSSL}
+         * {@linkplain SslContextBuilder#sslProvider(SslProvider) TLS/SSL protocol provider} via the tuner.
+         * Note that doing so adds a runtime dependency on
+         * <a href="https://netty.io/wiki/forked-tomcat-native.html">netty-tcnative</a>, which you must satisfy.
+         * <p>
+         * Notes:
+         * <ul>
+         *    <li>Netty {@link SslContext} may not examine some
+         *    {@linkplain Security security}/{@linkplain System#getProperties() system} properties that are used to
+         *    <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html#Customization">
+         *    customize JSSE</a>. Therefore, instead of using them you may have to apply the equivalent configuration programmatically
+         *    via the tuner, if both the {@link SslContextBuilder} and the TLS/SSL protocol provider of choice support it.
+         *    </li>
+         *    <li>Only {@link SslProvider#JDK} and {@link SslProvider#OPENSSL} TLS/SSL protocol providers are supported.
+         *    </li>
+         * </ul>
+         *
+         * @param nettySslContextTuner The tuner for an {@link SslContext}.
+         *                             The {@link SslContextBuilder} {@linkplain Consumer#accept(Object) supplied} to the tuner
+         *                             is created via {@linkplain SslContextBuilder#forClient()}.
+         * @return {@code this}.
+         *
+         * @since 4.3
+         */
+        public Builder applyToNettySslContext(final Consumer<? super SslContextBuilder> nettySslContextTuner) {
```

Comment:
We accept `Consumer<? super SslContextBuilder>` instead of accepting `SslContext` (as we do for the Java SE `SSLContext`) because there are server- and client-side implementations, and it seems better to create a builder for a client-side `SslContext` and give it to a user than to say in the docs that a user must start creating a context via `SslContextBuilder.forClient()`. We can discuss this and change if needed.

Reply:
Personally I'd rather keep both API's the similar and just provide the SslContext.  The documentation could highlight that it should be a client based context.  It seems that configuring this code is in the weeds already and advanced usage. That said you're better placed to determine the user pain of such an approach.

So, if you do wish to keep this more ergonomic approach then I suggest its updated to:

```
applyToSslContext(final Block<? super SslContextBuilder> sslContextBlock)
```

The `Netty` part is inferred from the type signature and the use of `Block` rather than `Consumer` is to follow suite with the rest of the configuration settings.

Reply:
I was on the fence. Since you prefer accepting `SslContext`, I changed the API.

Done.